tsl/src/hypercore/hypercore_handler.c

```diff
@@ -394,6 +395,34 @@ static bool hypercore_getnextslot_noncompressed(HypercoreScanDesc scan, ScanDire
 static bool hypercore_getnextslot_compressed(HypercoreScanDesc scan, ScanDirection direction,
 											 TupleTableSlot *slot);
 
+/*
+ * Skip scanning compressed data in a table scan.
+ *
+ * This function can be called on a scan descriptor to skip scanning of
+ * compressed data. Typically called directly after table_beginscan().
+ */
+void
+hypercore_scan_set_skip_compressed(TableScanDesc scan, bool skip)
```

Comment:
True, it is currently used only for testing. It seems reasonable, though, that the "rescan" API should support the ability to reset this flag so it is reasonable to be able to test it. Otherwise one would need to close and reopen a new scan. We could support  that if you think it is important to we don't support the rescan functionality.